src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectLogsActionTests.cs

```diff
@@ -0,0 +1,347 @@
+                options.FilterSpecs = configuration.FilterSpecs;
+
+                ICollectionRuleActionFactoryProxy factory;
+                Assert.True(host.Services.GetService<ICollectionRuleActionOperations>().TryCreateFactory(KnownCollectionRuleActions.CollectLogs, out factory));
+
+                using CancellationTokenSource endpointTokenSource = new CancellationTokenSource(CommonTestTimeouts.LogsTimeout);
+
+                EndpointInfoSourceCallback endpointInfoCallback = new(_outputHelper);
+                await using var source = _endpointUtilities.CreateServerSource(out string transportName, endpointInfoCallback);
+                source.Start();
+
+                AppRunner runner = _endpointUtilities.CreateAppRunner(transportName, TargetFrameworkMoniker.Net60); // Arbitrarily chose Net60; should we test against other frameworks?
+                runner.ScenarioName = TestAppScenarios.Logger.Name;
+
+                Task<IEndpointInfo> newEndpointInfoTask = endpointInfoCallback.WaitForNewEndpointInfoAsync(runner, CommonTestTimeouts.StartProcess);
+
+                await runner.ExecuteAsync(async () =>
+                {
+                    IEndpointInfo endpointInfo = await newEndpointInfoTask;
+
+                    ICollectionRuleAction action = factory.Create(endpointInfo, options);
+
+                    using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.LogsTimeout);
+
+                    CollectionRuleActionResult result;
+                    try
+                    {
+                        await action.StartAsync(cancellationTokenSource.Token);
+
+                        // Add a delay before the start of logging (this approach was first used in LogsTests.cs)
+                        await Task.Delay(TimeSpan.FromSeconds(3));
+
+                        await runner.SendCommandAsync(TestAppScenarios.Logger.Commands.StartLogging);
+
+                        result = await action.WaitForCompletionAsync(cancellationTokenSource.Token);
+                    }
+                    finally
+                    {
+                        await Tools.Monitor.DisposableHelper.DisposeAsync(action);
+                    }
+
+                    Assert.NotNull(result.OutputValues);
+                    Assert.True(result.OutputValues.TryGetValue(CollectionRuleActionConstants.EgressPathOutputValueName, out string egressPath));
+                    Assert.True(File.Exists(egressPath));
+
+                    using FileStream logsStream = new(egressPath, FileMode.Open, FileAccess.Read);
+                    Assert.NotNull(logsStream);
+
+                    await ValidateLogsEquality(logsStream, callback, logFormat);
+                });
+            });
+        }
+
+        private async Task ValidateLogsEquality(Stream logsStream, Func<ChannelReader<LogEntry>, Task> callback, LogFormat logFormat)
```

Comment:
If this is the same as the ValidatLogsEquality method in the FunctionalTests, then move it to TestCommon and have both sets of tests add this one.